chainer/dataset/dataset_mixin.py

```diff
@@ -0,0 +1,46 @@
+class DatasetMixin(object):
+
+    """Default implementation of dataset indexing.
+
+    DatasetMixin provides the :meth:`__getitem__` method using the per-example
+    addressing method :meth:`get_example` given by the implementation. This
```

Comment:
It is better to separate this sentence into two parts. For example, "DatasetMixin provides ...  method get_example. The concreate functionality of get_example is provided by the implementation."